SE-XXX Typed throws.md

```diff
@@ -616,15 +668,41 @@ catch {
 }
 ```
 
-### `rethrow` (generic errors in map, filter etc)
+### `rethrow` (generic errors in map, filter, etc)`
 
-`// TODO: Explain, merge with already explained part`
+While affecting to the `throws` keyword, there should be to have in consideration the `rethrows` clause when a type is present in the method signature, altough there is no impact over rethrows, as it inherits from its inner throwing type: 
 
-### `async` and `throws`
+```swift
+func foo<T>(_ block: () throws T -> Void) rethrows T
+```
 
-`// TODO: Maybe we can remove that section`

```

Comment:
What's the state of `async`. Is it going to be part of the language?

Reply:
In the future yes, will do, but we should not make our proposal over possible future language features. If they add it, as it send they'll do, we'll update the proposal